docs/intro-react.md

```diff
@@ -0,0 +1,476 @@
+
+function Cat() {
+
+  const [name, setName] = useState("What's my name again?");
+
+  return (
+    <View>
+      <Text>Hello, I am...{name}!</Text>
+      <TextInput
+        style={{
+          height: 40,
+          borderColor: 'gray',
+          borderWidth: 1
+        }}
+        onChangeText={
+          (text) => setName(text)
+        }
+      />
+    </View>
+  );
+}
+
+export default function Cafe() {
+  return (
+    <>
+      <Cat />
+      <Cat />
+    </>
+  );
+}
+```
+
+> See the `<>` and `</>` above? These bits of JSX are [fragments](https://reactjs.org/docs/fragments.html). Adjacent JSX elements must be wrapped in an enclosing tag. Fragments let you do that without nesting an extra, unnecessary wrapping element like `View`.
+
+First, you will want to import `useState` from React like so:
+
+```jsx
+import React, {useState} from 'react';
+```
+
+In this example, inside your `Cat` function, `useState` creates a `name` state variable:
+
+```jsx
+const [name, setName] = useState("What's my name again?");
+```
+
+Then you add `onChangeText` to the `<TextInput>`. This fires every time the `<TextInput>`‘s value changes (every time a user adds a character of text).
+
+```jsx
+onChangeText={(text) => setName(text)}
+```
+
+This uses the `setName` you defined with the `useState` Hook above to update your state variable `defaultText` every time `onChangeText` fires. You might set state when you have new data from the server, or from user input, or even from a timer.
+
+> You might’ve noticed that although `name` is a [const](https://developer.mozilla.org/Web/JavaScript/Reference/Statements/const), it is seemingly reassignable! What is happening is that every time a user inputs a letter into `<TextInput>`, `setName()` fires to update the state const `name`. When a state-setting function like `setName` is called, its component will re-render, running its function again. In this case `Cat` will re-render its `<Cat>`—discarding the old state const `name` in the process!
+
+<block class="classical syntax" />
+The older class components approach is a little different when it comes to state.
+
+```SnackPlayer name=State%20and%20Class%20Components
+import React, { Component } from 'react';
+import { Text, TextInput, View } from 'react-native';
+
+export class Cat extends Component {
+  state = {name: "What's my name again?!"}
+
+  render(){
+    return (
+      <View>
+        <Text>Hello, I am...{this.state.name}!</Text>
+        <TextInput
+          style={{
+            height: 40,
+            borderColor: 'gray',
+            borderWidth: 1
+          }}
+          onChangeText={
+            (text) => this.setState({name: text})
+          }
+        />
+      </View>
+    );
+  }
+}
+
+export default class Cafe extends Component {
+  render(){
+    return (
+      <>
+        <Cat />
+        <Cat />
+      </>
+    );
+  }
+}
+```
+
+In class components, state is stored in a `state` object:
```

Comment:
Notice how much simpler the class section looks when you switch. This is not the impression we want to make.

Reply:
Well, this is because we go into depth on what Hooks are and how this one works and the other example is there for reference. I can try to trim down the function component section and perhaps expand a bit on the class component